### PR TITLE
Handle Unicoded filenames add via SAB GUI (cherrypy)

### PR DESCRIPTION
**Pull request by sanderjo**
in SABnzbd's GUI, if you upload a NZB file with unicode in it, the filename gets garbled in SABnzbd.
the cause is in cherrypy
this PR is a workaround so that you can upload unicoded nzb filenames via the SAB GUI.

image

### CODE REVIEW

**Comment by sanderjo** [comment]
@Safihre ready for review

File: sabnzbd/encoding.py [hunk 1]
```diff
@@ -77,6 +77,16 @@ def correct_unknown_encoding(str_or_bytes_in: AnyStr) -> str:
             return str_or_bytes_in.decode(chardet.detect(str_or_bytes_in)["encoding"])
 
 
+def correct_cherrypy_encoding(somestring: str) -> str:
+    """convert somestring with seperate, individual chars (1-255) to valid string (with UTF8 encoding)"""
+    try:
+        correctedstring = somestring.encode("raw_unicode_escape").decode("utf8")
```
Comment (Safihre) [inline sabnzbd/encoding.py:83]: Let's just do `return` right here

File: sabnzbd/encoding.py [hunk 2]
```diff
@@ -77,6 +77,16 @@ def correct_unknown_encoding(str_or_bytes_in: AnyStr) -> str:
...
+    except:
+        # not possible to convert to UTF8, so don't change anything:
+        correctedstring = somestring
```
Comment (Safihre) [inline sabnzbd/encoding.py:86]: Same!

**Review by Safihre — CHANGES_REQUESTED** [review]
Small comment. And maybe put a reference to the issue in the PR text.

**Pull request merged**